### PR TITLE
feat(starfish): Fixes spans view operation, action, and domain filter dropdown items to match actual results

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Dropdown options for operation, action, and domain filter were not matching the actual unique values in the table result. This is because the query filters for the operation, action, and domain requests didnt match the filter for the table request.

### CODE REVIEW

File: static/app/views/starfish/utils/buildEventViewQuery.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import {ModuleName} from 'sentry/views/starfish/types';
+import {NULL_SPAN_CATEGORY} from 'sentry/views/starfish/views/webServiceView/spanGroupBreakdownContainer';
+
+const SPAN_FILTER_KEYS = [
+  'span.op',
+  'span.domain',
+  'span.action',
+  '!span.module',
+  '!span.category',
+];
```
Comment (DominikB2014) [inline static/app/views/starfish/utils/buildEventViewQuery.tsx:13]: You can use the `SpanMetricsFields`, similar to this refactor https://github.com/getsentry/sentry/pull/52657

Comment (edwardgou-sentry) [reply static/app/views/starfish/utils/buildEventViewQuery.tsx]: good call, refactored to use the new constants!

File: static/app/views/starfish/views/spans/selectors/actionSelector.tsx [hunk 2]
```diff
@@ -83,34 +83,21 @@ const LABEL_FOR_MODULE_NAME: {[key in ModuleName]: ReactNode} = {
...
-    } else if (spanCategory !== 'Other') {
-      queryConditions.push(`span.category:${spanCategory}`);
-    }
-  }
+  const query = buildEventViewQuery(
+    moduleName,
+    location,
+    undefined,
+    undefined,
+    spanCategory
```
Comment (DominikB2014) [inline static/app/views/starfish/views/spans/selectors/actionSelector.tsx:91]: Can definetely think about this in another PR, but I would vote to make a single object parameters in `buildEventViewQuery` vs multiple.
I think it looks a bit cleaner to omit a property of an object vs passing in undefined

File: static/app/views/starfish/views/spans/selectors/actionSelector.tsx [hunk 3]
```diff
@@ -83,34 +83,21 @@ const LABEL_FOR_MODULE_NAME: {[key in ModuleName]: ReactNode} = {
...
+  )
+    .filter(Boolean)
```
Comment (DominikB2014) [inline static/app/views/starfish/views/spans/selectors/actionSelector.tsx:93]: Is filtering by Boolean necessary here? I would find it a bit odd that `buildEventViewQuery` might return empty conditions and that isn't handled in `buildEventViewQuery` itself.

**Review by DominikB2014 — COMMENTED** [review]
I'm not sure if this is related to your changes, if not then code lgtm!.
But i'm noticing a bug.
When I have 'ALL' selected, and then I click on an operation, for example `db.redis`
image
Then the next time I click to select an operation, the other options disappear
image

I replicated this with the action and domain dropdowns too, and I can't replicate in prod

File: static/app/views/starfish/views/spans/selectors/actionSelector.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/views/starfish/views/spans/selectors/actionSelector.tsx]: Agreed, I've updated in this pr!

File: static/app/views/starfish/views/spans/selectors/actionSelector.tsx (hunk 3 above)
Comment (edwardgou-sentry) [reply static/app/views/starfish/views/spans/selectors/actionSelector.tsx]: Not necessary here, updated!

**Comment by edwardgou-sentry** [comment]
> I'm not sure if this is related to your changes, if not then code lgtm!. But i'm noticing a bug. When I have 'ALL' selected, and then I click on an operation, for example `db.redis` image Then the next time I click to select an operation, the other options disappear image
> 
> I replicated this with the action and domain dropdowns too, and I can't replicate in prod

Nice find, this is actually a bug. What's happening here is that the query for retrieving possible action/domain/op values is being filtered by the same action/domain/op selected on the page. Fixed this by excluding action/domain/op for each respective filter.

**Review by DominikB2014 — APPROVED** [review]
lgtm! Thanks for addressing the feedback

**Pull request merged**